p2p/service.go

```diff
@@ -175,18 +168,49 @@ func (s *Service) handleConn(conn network.Conn) {
 	}
 }
 
-// handleStream parses the message written to the data stream and calls the
-// associated message handler (non-status or status) based on message type
+// handleStream starts reading from the inbound message stream (substream with
+// a matching protocol id that was opened by the connected peer) and continues
+// reading until the inbound message stream is either closed or reset.
 func (s *Service) handleStream(stream network.Stream) {
+	ctx := context.Background()
+
+	// read from inbound stream
+	go s.readStream(ctx, stream)
```

Comment:
are you sure these needs its own goroutine? My expectation would be that this is called from a goroutine (as there are many simultaneous connections potentially happening). Removing it doesn't break tests, lmk what you think

Reply:
I am not sure. I think we can remove... Let me take another look.

Reply:
Yea, seems to be fine without it. Removed.